aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -21,9 +22,18 @@ namespace impl {
 // TODO: I'm not sure if this should live in this header or not; the operant
 // question is whether or not we have access to all the relevant TLS at this
 // point.
-static inline DispatchKey dispatchTypeId(DispatchKeySet ts) {
+static inline DispatchKey dispatchTypeId(
+    DispatchKeySet ts,
+    // This argument is used to mask out fallthrough keys, so we don't
+    // consider them for dispatch.  It's taken in the weird "inverted" form so
+    // that we can do it in one instruction (x & y) rather than two (x & ~y).
+    // These exclusions are NOT tracked in the TLS, but must be applied AFTER
+    // TLS, which is why you have to pass them in to this function (as opposed
+    // to just applying it to the input 'ts').
+    DispatchKeySet non_fallthrough_mask
```

Comment:
from a naming point of view, I wouldn't name this a mask. You're using it as a mask but that is an implementation detail that callers don't know about. The argument (from a callers perspective) is semantically a set of dispatch keys, namely the dispatch keys that don't have a fallthrough kernel registered. So maybe call it `backends_without_fallthrough` or something like that? That would also allow you to avoid or shorten the comment explaining that the mask is inverted.